pygfx/objects/_lights.py

```diff
@@ -288,16 +291,20 @@ def target(self, target):
         self._target = target
 
     def _gfx_update_uniform_buffer(self):
-        pos1 = self.get_world_position()
+        pos1 = self.world_transform.position
         pos2 = get_pos_from_camera_parent_or_target(self)
-        origin_to_target = Vector3().sub_vectors(pos2, pos1)
-        self._gfx_distance_to_target = origin_to_target.length()
+        origin_to_target = pos2 - pos1
+        self._gfx_distance_to_target = np.linalg.norm(origin_to_target)
         if self._gfx_distance_to_target > 0:
-            direction = origin_to_target.normalize()
+            direction = origin_to_target / self._gfx_distance_to_target
         else:
-            direction = Vector3(0, 0, -1)  # ill-defined direction -> look neg z-axis
-        self.uniform_buffer.data["direction"].flat = direction.to_array()
-        self.look_at(pos2)
+            direction = np.array(
+                (0, 0, -1), dtype=float
+            )  # ill-defined direction -> look neg z-axis
+        self.uniform_buffer.data["direction"].flat = direction
+        # note: we are doing computation in world space but then apply look_at
+        # in local space is this what we want?
+        self.transform.look_at(pos1 + direction)
```

Comment:
That makes sense. If it should do look-at in world space, then I will move the method to the world transform (and eventually out of the transform class and into the logic of WorldObject, because that's much more cohesive.